### PR TITLE
FLPATH-372: user-project endpoints 

### PR DESCRIPTION
**Pull request by RichardW98**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (FLPATH-xxxx)*:
Fixes #

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/project/controller/ProjectController.java [hunk 1]
```diff
@@ -104,4 +107,33 @@ public ResponseEntity<ProjectResponseDTO> getProjectById(@PathVariable UUID id)
...
+							schema = @Schema(implementation = ProjectUserRoleResponseDTO.class)) }),
+			@ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content),
+			@ApiResponse(responseCode = "404", description = "Not found", content = @Content) })
+	@DeleteMapping("/{id}/users")
+	public ResponseEntity<ProjectUserRoleResponseDTO> removeUsersFromProject(@PathVariable UUID id,
+			@RequestBody List<String> usernames) {
+		ProjectUserRoleResponseDTO projectUserRoleResponseDTO = projectService.removeUsersFromProject(id, usernames);
+		return ResponseEntity.ok(projectUserRoleResponseDTO);
+	}
+
```
Comment (pkliczewski) [inline workflow-service/src/main/java/com/redhat/parodos/project/controller/ProjectController.java:138]: What about update and list?

File: workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java [hunk 2]
```diff
@@ -48,4 +48,44 @@ public static String getUsername() {
 		return null;
 	}
 
+	public static String getEmail() {
```
Comment (pkliczewski) [inline workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java:51]: All the methods below use the same code as `getUsername`. Please fix

**Review by pkliczewski — COMMENTED** [review]
Please add tests. How are we going to manage role assignment on the UI?

File: workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java (hunk 2 above)
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/security/SecurityUtils.java:51]: yes, good catch! I will remove them

File: workflow-service/src/main/java/com/redhat/parodos/project/controller/ProjectController.java (hunk 1 above)
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/project/controller/ProjectController.java:138]: to list the users left in the project?  the response here will return all the updated data of this project where the given users are removed

File: workflow-service/src/main/java/com/redhat/parodos/project/dto/UserRoleRequestDTO.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserRoleRequestDTO {
+
+	String username;
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/project/dto/UserRoleRequestDTO.java:15]: private?

File: workflow-service/src/main/java/com/redhat/parodos/project/dto/UserRoleRequestDTO.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+	Role role;
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/project/dto/UserRoleRequestDTO.java:17]: private?

File: workflow-service/src/main/java/com/redhat/parodos/project/dto/ProjectUserRoleResponseDTO.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	private String projectName;
+
+	private List<UserRoleResponseDTO> userResponseDTOList;
+
+	@Builder
+	@Data
+	@AllArgsConstructor
+	@NoArgsConstructor
+	public static class UserRoleResponseDTO {
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/project/dto/ProjectUserRoleResponseDTO.java:49]: Can this class be defined in its own file?

File: workflow-service/src/main/java/com/redhat/parodos/project/repository/ProjectUserRoleRepository.java [hunk 6]
```diff
@@ -13,4 +13,6 @@ public interface ProjectUserRoleRepository extends JpaRepository<ProjectUserRole
 
 	List<ProjectUserRole> findByUserId(UUID userId);
 
+	List<ProjectUserRole> deleteAllByIdProjectIdAndIdUserIdIn(UUID projectId, List<UUID> userIds);
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/project/repository/ProjectUserRoleRepository.java:16]: there is no coverage for the new *Repository classes introduced in the previous PR. pls add.

File: workflow-service/src/main/java/com/redhat/parodos/project/service/ProjectServiceImpl.java [hunk 7]
```diff
@@ -118,4 +125,52 @@ public List<ProjectResponseDTO> getProjectsByUsername(String username) {
...
+				.map(userRoleMap -> ProjectUserRole.builder()
+						.id(ProjectUserRole.Id.builder().projectId(project.getId()).userId(userRoleMap.getKey().getId())
+								.roleId(userRoleMap.getValue().get().getId()).build())
+						.project(project).user(userRoleMap.getKey()).role(userRoleMap.getValue().get()).build())
+				.collect(Collectors.toSet());
+
+		projectUserRoleRepository.deleteAllByIdProjectIdAndIdUserIdIn(project.getId(),
+				projectUserRoles.stream().map(projectUserRole -> projectUserRole.getUser().getId()).toList());
+		project.setProjectUserRoles(projectUserRoles);
+		projectUserRoleRepository.saveAll(projectUserRoles);
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/project/service/ProjectServiceImpl.java:147]: Can you explain the delete & save here?

File: workflow-service/src/main/java/com/redhat/parodos/user/service/UserService.java [hunk 8]
```diff
@@ -36,4 +37,6 @@ public interface UserService {
 
 	User getUserEntityByUsername(String username);
 
+	List<User> findAllUserEntitiesByUsernameIn(List<String> usernames);
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/user/service/UserService.java:40]: test?
also, can we omit `Entities` from the method name?

File: workflow-service/src/main/java/com/redhat/parodos/project/dto/ProjectUserRoleResponseDTO.java (hunk 5 above)
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/project/dto/ProjectUserRoleResponseDTO.java:49]: yes sure

File: workflow-service/src/main/java/com/redhat/parodos/project/service/ProjectServiceImpl.java (hunk 7 above)
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/project/service/ProjectServiceImpl.java:147]: the mapping between projectUserrole and project is not bidrectional. what I updated in project/projectUserrole will be not reflected to the other in db. so I have to delete and refill. this is the known limition of embedded collection

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**